### PR TITLE
feat!: add ATLAS_OPTIONS and remove i18n-merge.js | FC-0012

### PR DESCRIPTION
**Pull request by OmarIthawi**
This pull request is part of the [FC-0012 project](https://openedx.atlassian.net/l/cp/XGS0iCcQ) which implements the [Translation Infrastructure update OEP-58](https://docs.openedx.org/en/latest/developers/concepts/oep58.html).

### CODE REVIEW

**Review by regisb — COMMENTED** [review]
Please update the README instructions and add a changelog entry :)

**Comment by OmarIthawi** [comment]
> Please update the README instructions and add a changelog entry :)

I hear that as a "yes" :) 

Will do. Thanks @regisb.

**Comment by regisb** [comment]
Oh wait, this PR should be on top of the nightly branch, since we don't aim to support Quince.

**Comment by OmarIthawi** [comment]
That's right @regisb. I meant to open it on `nightly`.

**Comment by OmarIthawi** [comment]
@arbrandes @regisb this is now ready.

**Review by regisb — APPROVED** [review]
This looks great to me! What do you think @arbrandes?

**Review by arbrandes — APPROVED** [review]
This is awesome.  Thanks @OmarIthawi!

File: tutormfe/templates/mfe/build/mfe/Dockerfile [hunk 1]
```diff
@@ -65,12 +47,11 @@ RUN --mount=type=cache,target=/root/.npm,sharing=shared npm clean-install --no-a
 {{ patch("mfe-dockerfile-post-npm-install") }}
 {{ patch("mfe-dockerfile-post-npm-install-{}".format(app_name)) }}
 COPY --from={{ app_name }}-src / /openedx/app
-COPY --from={{ app_name }}-i18n /openedx/app/src/i18n/messages /openedx/app/src/i18n/messages
 
 # Whenever a new MFE supports Atlas, it should be added to this list.
 # When all MFEs support Atlas, this if-statement should be removed.
 {% if app_name in ["communications"] %}
-RUN make OPENEDX_ATLAS_PULL=true pull_translations
+RUN make OPENEDX_ATLAS_PULL=true ATLAS_OPTIONS="--repository={{ ATLAS_REPOSITORY }} --branch={{ ATLAS_REVISION }} {{ ATLAS_OPTIONS }}" pull_translations
```
[the remark this answers is not in the document]
Comment (OmarIthawi) [reply tutormfe/templates/mfe/build/mfe/Dockerfile:54]: > now that we're moving to using revisions (tags/shas) instead of branches when pulling with atlas, does it make sense to still use --branch here?

@brian-smith-tcril the `ATLAS_REVISION` choice is a bit debatable but I've accepted [Regis feedback](https://github.com/overhangio/tutor/pull/964#discussion_r1429673724), but we ended up using it for Tutor even though it's still technically the same `--branch` argument.

Currently the `--branch` argument supports both tags and branches, but not commit SHA due to `git clone --branch` limitation:

```
$ git clone --help | grep -A2 '.-branch '
       -b <name>, --branch <name>
           Instead of pointing the newly created HEAD to the branch pointed to by the cloned repository’s HEAD, point to <name> branch
           instead. In a non-bare repository, this is the branch that will be checked out.  --branch can also take tags and detaches the
           HEAD at that commit in the resulting repository.
```

We haven't added support yet for all git revisions:

 - https://github.com/openedx/openedx-atlas/issues/48

> If not, it might make sense to update the atlas CLI to use a new argument name before merging this.

In princinple I agree that adding the `--revision` option is better before merging this pull request, but it'll require a backlog of small pull requests to be merged across multiple repositories which could take a significant time.

Once this is done, I believe we can make it backward compatible way by adding a new `--revision` option which uses `--branch` as an alias.

Therefore, I think keeping `ATLAS_REVISION` is somewhat acceptable even though it doesn't match `atlas pull --branch` name.

To summarize, if adding the `--revision` option to `atlas` is a priority, I believe it should be done after merging this pull request since we're almost done and I'd like to keep a good window with Atlas being merged before Redwood is realesed.

[the remark this answers is not in the document]
Comment (brian-smith-tcril) [reply tutormfe/templates/mfe/build/mfe/Dockerfile:54]: Are we using `--branch` anywhere else yet? If not, could we just make a quick PR to `atlas` changing the current `--branch` argument to `--revision` and update it to support pulling by `sha` after?

[the remark this answers is not in the document]
Comment (OmarIthawi) [reply tutormfe/templates/mfe/build/mfe/Dockerfile]: @brian-smith-tcril The rename has been done. Thanks for bringing it up.


@arbrandes This is now ready for merge.

**Comment by OmarIthawi** [comment]
@arbrandes All notes have been address, this pull request is now ready.

cc: @brian-smith-tcril @regisb

**Review by regisb — APPROVED** [review]
LGTM!
